examples/release_highlights/plot_release_highlights_0_24_0.py

```diff
@@ -0,0 +1,225 @@
+# Searching parameter space with successive halving
+# -------------------------------------------------
+# Successive halving, a state of the art method, is now available to
+# explore the space of the parameters and identify their best combination.
+# It is still experimental, in order to import it `enable_halving_search_cv`
+# should be imported from `experimental`.
+# :class:`~sklearn.model_selection.HalvingGridSearchCV` and
+# :class:`~sklearn.model_selection.HalvingRandomSearchCV` can be
+# used as drop-in replacement for
+# :class:`~sklearn.model_selection.GridSearchCV` and
+# :class:`~sklearn.model_selection.RandomizedSearchCV`.
+# Successive halving is an iterative selection process. The first iteration is
+# run with a small amount of resources, by resource meaning typically the
+# number of training samples, but also arbitrary numeric parameters such
+# as `n_estimators` in a random forest. Only some of the parameter candidates
+# are selected for the next iteration, with an increasing size of the
+# allocated resources.
+# As illustrated in the figure below,
+# only a subset of candidates will last until the end of the iteration process.
+# Read more in the :ref:`User Guide <successive_halving_user_guide>`.
+# 
+# .. figure:: ../model_selection/images/sphx_glr_plot_successive_halving_iterations_001.png
+#   :target: ../model_selection/plot_successive_halving_iterations.html
+#   :align: center
+
+import numpy as np
+from scipy.stats import randint
+from sklearn.experimental import enable_halving_search_cv  # noqa
+from sklearn.model_selection import HalvingRandomSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+
+rng = np.random.RandomState(0)
+
+X, y = make_classification(n_samples=700, random_state=rng)
+
+clf = RandomForestClassifier(n_estimators=20, random_state=rng)
+
+param_dist = {"max_depth": [3, None],
+              "max_features": randint(1, 11),
+              "min_samples_split": randint(2, 11),
+              "bootstrap": [True, False],
+              "criterion": ["gini", "entropy"]}
+
+rsh = HalvingRandomSearchCV(estimator=clf, param_distributions=param_dist,
+                            factor=2, random_state=rng)
+rsh.fit(X, y)
+rsh.best_params_
+
+##############################################################################
+# Native support for categorical features in HistGradientBoosting
+# ---------------------------------------------------------------
+# :class:`sklearn.ensemble.HistGradientBoostingClassifier` and
+# :class:`sklearn.ensemble.HistGradientBoostingRegressor` now have native
+# support for categorical features: they can consider splits on non-ordered,
+# categorical data. Read more in the :ref:`User Guide
+# <categorical_support_gbdt>`.
+#
+# .. figure:: ../ensemble/images/sphx_glr_plot_gradient_boosting_categorical_001.png
+#   :target: ../ensemble/plot_gradient_boosting_categorical.html
+#   :align: center
```

Comment:
IMO the plot isn't convincing enough on its own without the discussion from the example, because the Ordinal method and  the native one are so close (also fitting time can vary quite significantly from one execution to the other). Maybe we can just have a really small snippet to illustrate the new parameter instead of the plot, as is done briefly in the UG

Reply:
Since the API is not user-friendly at the moment (need for ColumTransformer + OrdinalEncoding preprocessing + mask that depends on the reordering of the features caused by the ColumTransformer) we cannot show a simple code snippet that is representative of the actual user experience. So I think it's better to link to the example to give a guideline on how to use this for now and maybe mentioning that we plan to improve user friendliness in the future.

I find the benchmark interesting. Many people are not aware that ordinal encoding alone is already a very good way to preprocess categorical features for trees. We can add sentence explaining that the native categorical handling is typically as fast as Ordinal Encoding while even more expressive than both `OneHotEncoding` and `OrdinalEncoding` and can lead to shallower models that predict faster when combined with early stopping.

Reply:
I think this section would benefit from being extended a bit:

a- with a short analysis of the important points shown by the plot;
b- a note about the need for preprocessing.

For a. I would say that this plot shows that the new native support for categorical features leads to model that are almost as fast to train as model trained on arbitrarily integer encoded categorical values (`OrdinalEncoder`).

For b. we could say that it is important to note that to use the new `categorical_features`, it is still required to preprocess the data with a pipeline as demonstrated in the example linked by clicking on the figure above. The `HistGradientBoosting` will be further improved in the next release to make it more natural to use this new feature.

We could optionally link to https://github.com/scikit-learn/scikit-learn/issues/18894 for those interested in following the progress on that specific point (planned future improvement in the usability).

Reply:
> This new way to handle categorical variables can further lead to shallower models that predict faster when combined with early stopping

Do we really need to specify "when combined with early stopping"? The benefits of the native handling seem unrelated to ES to me

Reply:
Some details added, without reference to early stopping.